lib/src/migrators/module.dart

```diff
@@ -527,4 +545,88 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
     }
     return _namespaces[node.span.sourceUrl];
   }
+
+  /// Disallows @use after @at-root rules.
+  @override
+  void visitAtRootRule(AtRootRule node) {
+    _useAllowed = false;
+    super.visitAtRootRule(node);
+  }
+
+  /// Disallows @use after at-rules other than @charset.
+  @override
+  void visitAtRule(AtRule node) {
+    if (node.name.asPlain == 'charset') _useAllowed = false;
```

Comment:
`@charset` rules are discarded during parse-time, so you don't need to worry about them here.